### PR TITLE
Implement two new alert functions

### PR DESCRIPTION
**Pull request by HoOngEe**
Implemented two alert functions. One is alerted when the best block's view value is higher than 2. The other is alerted when the precommit state is changed. The change occurs when some nodes start not to precommit or when these nodes restart to precommit so that all nodes are validating.

### CODE REVIEW

**Comment by sgkim126** [comment]
I think `Refactor` is not enough to explain the commit. Please write down the reason that you refactored the code.

File: monitor/index.ts [hunk 1]
```diff
@@ -2,21 +2,40 @@ import { SDK } from "codechain-sdk";
...
     if (prevBestBlockNumber === currentBestBlockNumber) {
-      sendNotice(chainErrors.codechainDeath);
+      sendNotice(new chainErrors.CodeChainDeath());
     }
     prevBestBlockNumber = currentBestBlockNumber;
+  };
+})();
+
+async function checkView(sdk: SDK) {
+  const viewAlertLevel = 3;
```
Comment (sgkim126) [inline monitor/index.ts]: Would you make it configurable?

File: monitor/index.ts [hunk 2]
```diff
@@ -2,21 +2,40 @@ import { SDK } from "codechain-sdk";
...
+  const bestBlockNumber = await sdk.rpc.chain.getBestBlockNumber();
+  const bestBlock = await sdk.rpc.chain.getBlock(bestBlockNumber);
+  if (bestBlock) {
```
Comment (sgkim126) [inline monitor/index.ts]: Is it okay that the best block is null?

File: monitor/index.ts (hunk 1 above)
Comment (HoOngEe) [reply monitor/index.ts]: Now viewAlertLevel is loaded from configuration

File: monitor/index.ts (hunk 2 above)
Comment (HoOngEe) [reply monitor/index.ts]: I added another alert.

File: monitor/Errors.ts [hunk 3]
```diff
@@ -3,7 +3,22 @@ export interface CodeChainAlert {
...
-  title: "[CodeChain Death Confirmation]",
-  content: "CodeChain didn't renew the best block number for 1 hour."
-};
\ No newline at end of file
+export class CodeChainDeath implements CodeChainAlert {
+  public title: string;
+  public content: string;
+
+  constructor() {
+    this.title = "[CodeChain Death Confirmation]";
```
Comment (sgkim126) [inline monitor/Errors.ts]: Would you add the network id on the title?
I want to show the title looks like `[cc network Death Confirmation]` or `[wc network Death Confirmation]`

File: monitor/Errors.ts [hunk 4]
```diff
@@ -3,7 +3,22 @@ export interface CodeChainAlert {
...
+    this.content = "CodeChain didn't renew the best block number for 1 hour.";
+  }
+}
+
+export class ViewTooHigh implements CodeChainAlert {
+  public title: string;
+  public content: string;
+
+  constructor(view: number) {
+    this.title = "[CodeChain View Too High]";
```
Comment (sgkim126) [inline monitor/Errors.ts]: ditto

File: monitor/Errors.ts (hunk 3 above)
Comment (HoOngEe) [reply monitor/Errors.ts]: prefix added

File: monitor/index.ts [hunk 5]
```diff
@@ -1,30 +1,80 @@
...
-import { getConfig } from "./util";
-
-async function main() {
-  const rpcUrl = getConfig<string>("rpc_url");
-  const networkId = getConfig<string>("network_id");
+import { getConfig, unsetBitIndices } from "./util";
 
-  const sdk = new SDK({ server: rpcUrl, networkId });
-  const emailClient = new EmailClient("");
+const targetEmail = "devop@kodebox.io";
```
Comment (sgkim126) [inline monitor/index.ts]: Please make the email destination configurable.

Comment (HoOngEe) [reply monitor/index.ts]: Now the target email address is configurable.

**Pull request merged**